processing/src/main/java/io/druid/query/aggregation/LastAggregatorFactory.java

```diff
@@ -0,0 +1,335 @@
+import io.druid.segment.ColumnSelectorFactory;
+import io.druid.segment.ObjectColumnSelector;
+import io.druid.segment.column.Column;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+
+public class LastAggregatorFactory extends AggregatorFactory
+{
+  private static final byte CACHE_TYPE_ID = 0x11;
+  private static final String TYPE_LONG = "long";
+  private static final String TYPE_DOUBLE = "double";
+
+  private final String fieldName;
+  private final String name;
+  private final String value;
+
+  @JsonCreator
+  public LastAggregatorFactory(
+      @JsonProperty("name") String name,
+      @JsonProperty("fieldName") final String fieldName,
+      @JsonProperty("value") String value
+  )
+  {
+    Preconditions.checkNotNull(name, "Must have a valid, non-null aggregator name");
+    Preconditions.checkNotNull(fieldName, "Must have a valid, non-null fieldName");
+    Preconditions.checkArgument(
+        value.equals(TYPE_LONG) || value.equals(TYPE_DOUBLE),
+        "Must have a valid, non-null type"
+    );
+
+    this.name = name;
+    this.fieldName = fieldName;
+    this.value = value;
+  }
+
+  @Override
+  public Aggregator factorize(ColumnSelectorFactory metricFactory)
+  {
+    if (value.equals(TYPE_LONG)) {
+      return new LongLastAggregator(
+          name, metricFactory.makeLongColumnSelector(fieldName),
+          metricFactory.makeLongColumnSelector(Column.TIME_COLUMN_NAME)
+      );
+    } else if (value.equals(TYPE_DOUBLE)) {
+      return new DoubleLastAggregator(
+          name, metricFactory.makeFloatColumnSelector(fieldName),
+          metricFactory.makeLongColumnSelector(Column.TIME_COLUMN_NAME)
+      );
+    }
+    throw new IAE("undefined type");
+  }
+
+  @Override
+  public BufferAggregator factorizeBuffered(ColumnSelectorFactory metricFactory)
+  {
+    if (value.equals(TYPE_LONG)) {
+      return new LongLastBufferAggregator(
+          metricFactory.makeLongColumnSelector(Column.TIME_COLUMN_NAME),
+          metricFactory.makeLongColumnSelector(fieldName)
+      );
+    } else if (value.equals(TYPE_DOUBLE)) {
+      return new DoubleLastBufferAggregator(
+          metricFactory.makeLongColumnSelector(Column.TIME_COLUMN_NAME),
+          metricFactory.makeFloatColumnSelector(fieldName)
+      );
+    }
+    throw new IAE("undefined type");
+  }
+
+  @Override
+  public Comparator getComparator()
+  {
+    return new Comparator()
+    {
+      @Override
+      public int compare(Object o1, Object o2)
+      {
+        return Longs.compare(((SerializablePair<Long, Object>) o1).lhs, ((SerializablePair<Long, Object>) o2).lhs);
+      }
+    };
+  }
+
+  @Override
+  public Object combine(Object lhs, Object rhs)
+  {
+    return (((SerializablePair<Long, Object>) lhs).lhs > ((SerializablePair<Long, Object>) rhs).lhs) ? lhs : rhs;
+  }
+
+  @Override
+  public AggregatorFactory getCombiningFactory()
```

Comment:
Hmm, this is a bit weird, but I guess you're trying to avoid having both a LongLastAggregator and LongLastCombiningAggregator. I can't think of a better way to structure that right now and I'm not sure which of those two options I like better…